omegaconf/dictconfig.py

```diff
@@ -147,8 +147,26 @@ def _validate_merge(self, key: Any, value: Any) -> None:
         self._validate_set_merge_impl(key, value, is_assign=False)
 
     def _validate_set(self, key: Any, value: Any) -> None:
+        self._validate_set_value_node(key, value)
         self._validate_set_merge_impl(key, value, is_assign=True)
 
+    def _validate_set_value_node(self, key: Any, value: Any) -> None:
```

Comment:
why is this a new function and not in _validate_set_merge_impl? (which is supposed to validate both set and assign based on the is_assign flag)

Reply:
I find it confusing to stack different types of validation on the same function without splitting into different functions. I think this way it's more readable than hacking my way trough that big _validate_set_merge_impl function.

Reply:
even if I agree with this, now you have:
1. function that does both.
2. another function that does one.

Adding a new function without considering the existing design is not ok.

Reply:
Please combine it into _validate_set_merge_impl, we can have a followup refactoring to make the code easier to read later.